src/Arcus.Security.Tests.Integration/DockerSecrets/SecretStoreBuilderExtensionTests.cs

```diff
@@ -0,0 +1,85 @@
+﻿using Arcus.Security.Core;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Hosting;
+using System;
+using System.IO;
+using System.Threading.Tasks;
+using Arcus.Security.Providers.DockerSecrets;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace Arcus.Security.Tests.Integration.DockerSecrets
+{
+    public class SecretStoreBuilderExtensionTests : IntegrationTest, IDisposable
+    {
+        private readonly string _secretLocation = Path.Combine(Path.GetTempPath(), "dockersecretstests");
+
+        public SecretStoreBuilderExtensionTests(ITestOutputHelper testOutput) : base(testOutput)
+        {
+            Directory.CreateDirectory(_secretLocation);
+        }
+
+        [Fact]
+        public async Task AddDockerSecrets_WithPath_ResolvesSecret()
+        {
+            // Arrange
+            var expectedValue = Guid.NewGuid().ToString();
+            var secretKey = "MySuperSecret";
+            await SetSecretAsync(secretKey, expectedValue);
+
+            var hostBuilder = new HostBuilder();
+
+            // Act
+            hostBuilder.ConfigureSecretStore((config, stores) => stores.AddDockerSecrets(_secretLocation));
+
+            // Assert
+            IHost host = hostBuilder.Build();
+            var secretProvider = host.Services.GetRequiredService<ISecretProvider>();
+
+            string actualValue = await secretProvider.GetRawSecretAsync(secretKey);
+            Assert.Equal(expectedValue, actualValue);
+        }
+
+        [Fact]
+        public async Task DockerSecretsProvider_ReturnsNull_WhenSecretNotFound()
```

Comment:
I'm sorry for the thorough comments. I was thinking here. Maybe also a test where we set a secret to a docker secret file but request a secret with a name that isn't present in that secret file, so we can check that the selective searching is correct.

Reply:
I can do that, somewhere later this week probably.

Reply:
I've changed the existing testcase.   In that test, I just make sure that a secret exists.

Reply:
Yeah, but maybe both are intressting. Right? Checking that the secret provider works both for an existing Docker secrets file and without.

Reply:
I don't see much difference ?  Each secret has it's own file.  So if I want to retrieve a secret that doesn't exist, the file will not be there.
I don't see the added value of having 2 similar tests:  one where we have no secrets (and thus no secretfiles) and a test where we do have some secret files, but want to retrieve a secret that doesn't exist

Reply:
Yeah, I think it's valuable bc you can prove that with or without the file it results in the same outcome. And it doesn't retrieve another file or the wrong one. And it doesn't affect the way the provider works. Maybe I would write something that would write a number of random secrets and retrieve a subset of them , but that would be a property test and not an integration test so this is fine.
But, maybe it's not necessary. I'll resolve this.